test/test-assert.h

```diff
@@ -22,8 +22,24 @@ class assertion_failure : public std::logic_error {
 
 void assertion_failure::avoid_weak_vtable() {}
 
+inline void throw_assertion_failure (const char *message)
+{
+#  if defined(__GNUC__) && __GNUC__ >= 6
+     // Avoid warnings when FMT_ASSERT is used in a destructor.
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wterminate"
+#  endif
+
+  throw assertion_failure(message);
+
+#  if defined(__GNUC__) && __GNUC__ >= 6
+#    pragma GCC diagnostic pop
+#  endif
+}
```

Comment:
Do you need both the function and suppression? gcc doesn't seem to warn in this case: https://godbolt.org/z/PYzx9h

Reply:
Thanks for your review. I've pushed a couple of commits which address your comments.